### PR TITLE
feat: add long side + tests

### PR DESCRIPTION
**Pull request by hellodespereaux**

### CODE REVIEW

File: contracts/OptionScalp.sol [hunk 1]
```diff
@@ -273,188 +331,275 @@ Pausable {
...
-    function claimCollateral(uint amount) 
-    public {
-        require(msg.sender == address(scalpLp), "Only Scalp LP contract can claim collateral");
-        quote.transfer(msg.sender, amount);
+    /// @notice Returns whether an open position is liquidatable
+    function isLiquidatable(uint256 id) public view returns (bool) {
+        return int256(scalpPositions[id].margin) + calcPnl(id) <=
+            int256(minimumAbsoluteLiquidationThreshold) *
+                int256(scalpPositions[id].positions);
+    }
```
Comment (0xYYC) [inline contracts/OptionScalp.sol:520]: Please check if both side decimals compatible

File: contracts/OptionScalp.sol [hunk 2]
```diff
@@ -273,188 +331,275 @@ Pausable {
...
         require(scalpPositions[id].isOpen, "Invalid position ID");
+
         require(
-            IERC721(scalpPositionMinter).ownerOf(id) == msg.sender, 
-            "Sender must be position owner"
+            IERC721(scalpPositionMinter).ownerOf(id) == msg.sender ||
+                scalpPositions[id].openedAt + scalpPositions[id].timeframe >=
+                block.timestamp,
+            "Sender must be position owner or position must be expired"
         );
```
Comment (0xYYC) [inline contracts/OptionScalp.sol:347]: Two questions :
1,The message should be position must be expired or not expired yet ? ( As scalpPositions[id].openedAt + scalpPositions[id].timeframe >=
                block.timestamp mean postion not expired .
2) The logic is a bit loose as if the block.timestamp satisfied condition , anyone can close others position ?
If you want to add the logic that either the owner can close or keeper can close please add it .

File: contracts/OptionScalp.sol (hunk 1 above)
Comment (hellodespereaux) [reply contracts/OptionScalp.sol:520]: fixed, now both ie6

Comment (hellodespereaux) [reply contracts/OptionScalp.sol:522]: done

File: contracts/OptionScalp.sol (hunk 2 above)
Comment (hellodespereaux) [reply contracts/OptionScalp.sol:347]: The message is correct I think

Either you are the owner and you can close whatever or your position is expired and anybody can close it

Comment (hellodespereaux) [reply contracts/OptionScalp.sol:347]: We shouldn't have a whitelisted keeper, we should promote the idea of having multiple keepers written in multiple languages on multiple instances to be sure positions are closed asap

We may want to give a fee to the keeper which closes the position

Comment (0xYYC) [reply contracts/OptionScalp.sol:351]: scalpPositions[id].openedAt + scalpPositions[id].timeframe >= block.timestamp 

mean the position must be not expired yet ?

File: contracts/OptionScalp.sol [hunk 3]
```diff
@@ -273,188 +359,209 @@ Pausable {
...
+    /// @param id Identifier of the position
+    function getLiquidationPrice(uint256 id) public view returns (uint256 price) {
+        int256 threshold = int256(scalpPositions[id].margin) - (int256(minimumAbsoluteLiquidationThreshold) * int256(scalpPositions[id].positions)) / 10**8;
 
-        int pnl = (int)(finalSize - scalpPositions[id].size);
-        scalpLp.unlockLiquidity(scalpPositions[id].size);
-        if (pnl > 0) {
-            quote.transfer(positionOwner, (uint)((int)(scalpPositions[id].margin) + pnl));
+        if (scalpPositions[id].isShort) {
+          price = (divisor * (scalpPositions[id].size) / scalpPositions[id].positions) + (divisor * (uint(threshold) * 10 ** 2) / scalpPositions[id].positions); // ie8
```
Comment (0xYYC) [inline contracts/OptionScalp.sol:460]: Agree with the formulae , but will it be simpler to replace (divisor * (scalpPositions[id].size) / scalpPositions[id].positions) by scalpPositions[id].entry ?

Comment (hellodespereaux) [reply contracts/OptionScalp.sol:460]: yes kek

Comment (hellodespereaux) [reply contracts/OptionScalp.sol:460]: done

**Pull request merged**